src/randombytes.rs

```diff
@@ -41,6 +46,35 @@ pub fn randombytes_uniform(upper_bound: u32) -> u32 {
     unsafe { ffi::randombytes_uniform(upper_bound) }
 }
 
+new_type! {
+    /// `Seed` bytes for the deterministic random functions
+    secret Seed(SEEDBYTES);
```

Comment:
Why secret?

Reply:
It has to be if you want to use it to generate keys and such.

Reply:
I debated using a `Nonce` type but semantically it is a secret given that the RNG is purely deterministic given knowledge of the initial seed.

Reply:
Why not “pk”?

Reply:
I'm not sure that I follow. Do you mean `secret` vs `public` in `new_type` or do you mean as a naming convention?

It's closer to a secret than it is to a public key IMO. As the libsodium docs state:

> the randombytes_buf_deterministic function stores size bytes into buf indistinguishable from random bytes without knowing seed.

> For a given seed, this function will always output the same sequence. size can be up to 2^38 (256 GB).

It goes on to state that:

> This function is mainly useful for writing tests, and was introduced in libsodium 1.0.12

but using a PRNG with a secret seed and relying on the seed remaining secret isn't unheard of.

Reply:
I see.